### PR TITLE
feat: migrate policy to support messages

### PR DESCRIPTION
**Pull request by ytvnr**
BREAKING CHANGE: this version is using the latest dependencies introduced by gravitee 4.0

**Issue**

https://gravitee.atlassian.net/browse/APIM-1624

---
**Gravitee.io Automatic Deployment**

🚀 A prerelease version of this package has been published on Gravitee's private artifactory, you can:
 - use it directly by updating your project with version: `2.0.0`
 - download it from Artifactory [here](https://odbxikk7vo-artifactory.services.clever-cloud.com/gravitee-snapshots/com/graviteesource/policy/gravitee-policy-xml-json/2.0.0/gravitee-policy-xml-json-2.0.0.zip)

### CODE REVIEW

File: .circleci/config.yml
Comment (leleueri) [inline .circleci/config.yml:1]: Can you please update all the config to take into account alpha branchs during release ? (see https://github.com/gravitee-io/gravitee-policy-transformheaders/blob/master/.circleci/config.yml#L25 as an example)

**Review by leleueri — CHANGES_REQUESTED** [review]
LGTM in term of code but can you please adapt the plugin.properties file to define for each type of API on which phase it is accessible?

```
proxy=REQUEST,RESPONSE 
message=MESSAGE_REQUEST,MESSAGE_RESPONSE
```

cf slab page about  v4 "Conceptual changes"

File: .circleci/config.yml
Comment (ytvnr) [reply .circleci/config.yml:1]: good catch, I totally forgot those things.

for the Breaking change, it is indicated in the commit, but I can add it to the PR description also

**Review by leleueri — APPROVED** [review]

**Pull request merged**